### PR TITLE
VIMC-4594: orderly runner will only start if orderly root is version controlled

### PR DESCRIPTION
**Pull request by r-ash**

### CODE REVIEW

File: tests/testthat/helper-orderly.server.R [hunk 1]
```diff
@@ -125,6 +124,30 @@ orderly_path_orderly_run_rds <- orderly:::path_orderly_run_rds
 orderly_git_checkout_branch <- orderly:::git_checkout_branch
 ## nolint end
 
+orderly_git_example <- function(name, path = tempfile(), testing = FALSE) {
```
Comment (r-ash) [inline tests/testthat/helper-orderly.server.R:127]: Maybe we should move this into `orderly`? I can see updating `orderly_example` to have a `git` arg to initialise with git.

**Comment by r-ash** [comment]
When pinning orderly-web to this branch all their tests are still passing https://github.com/vimc/orderly.server/pull/55 so I think their setup must be using a git enabled repo already

File: tests/testthat/helper-orderly.server.R (hunk 1 above)
Comment (richfitz) [reply tests/testthat/helper-orderly.server.R:127]: yes, good idea. it's possibly where it came from in the first place?

File: R/main.R [hunk 2]
```diff
@@ -46,7 +46,7 @@ Options:
 main_worker <- function(args = commandArgs(TRUE)) {
   # nocov start
   args <- main_worker_args(args)
-  wait_for_go_signal(path, args$go_signal)
+  wait_for_go_signal(NULL, args$go_signal)
```
Comment (richfitz) [inline R/main.R:49]: surprised about this change, how did this work before?

**Review by richfitz — CHANGES_REQUESTED** [review]
this looks good once conflicts done

File: R/main.R (hunk 2 above)
Comment (r-ash) [reply R/main.R:49]: So I think the test cases meant that `path` was never evaluated. As the test `!grepl("^(/|[A-Z][a-z]:)", go_signal) && !is.null(path)` the `!grepl("^(/|[A-Z][a-z]:)", go_signal)` part is FALSE and so `path` never gets used

**Review by richfitz — APPROVED** [review]

**Pull request merged**